x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -30,6 +30,81 @@ interface ColumnChange {
   shouldResetLabel?: boolean;
 }
 
+interface ColumnCopy {
+  layer: IndexPatternLayer;
+  columnId: string;
+  sourceColumn: IndexPatternColumn;
+  sourceColumnId: string;
+  shouldDeleteSource?: boolean;
+  indexPattern: IndexPattern;
+}
+
+export function copyColumn({
+  layer,
+  columnId,
+  sourceColumn,
+  sourceColumnId,
+  shouldDeleteSource,
+  indexPattern,
+}: ColumnCopy): IndexPatternLayer {
+  let newColumns = {
+    ...layer.columns,
+    [columnId]: { ...sourceColumn },
+  };
+
+  const operationDefinition = operationDefinitionMap[sourceColumn.operationType];
+  const newIds: string[] = [];
+  if (operationDefinition.input === 'fullReference' && 'references' in sourceColumn) {
+    sourceColumn?.references.forEach((ref) => {
+      const newId = generateId();
+      newIds.push(newId);
+      const refId = Object.keys(layer.columns).find((c) => c === ref);
+      if (refId) {
+        newColumns = {
+          ...newColumns,
+          [newId]: { ...layer.columns[refId] },
+          [columnId]: { ...sourceColumn, references: [newId] },
+        };
+      }
+    });
+  }
+
+  const newColumnOrder = [...layer.columnOrder];
+
+  if (shouldDeleteSource) {
+    const sourceIndex = newColumnOrder.findIndex((c) => c === sourceColumnId);
```

Comment:
it looks like this can be replaced by the `deleteColumn` in `onMoveCompatible` (see previous comment)